File: app/src/main/java/com/dumuzeyn/mp3player/SongsMenuRenderer.java
package com.dumuzeyn.mp3player;

final class SongsMenuRenderer implements MenuRenderer {
    private final MainActivityCore host;

    SongsMenuRenderer(MainActivityCore host) {
        this.host = host;
    }

    void loadSongs() {
        host.libraryState.tracks.clear();
        host.libraryState.tracks.addAll(TrackStore.load(host));
    }

    @Override
    public boolean needsMiniSpacer() {
        return false;
    }

    @Override
    public void render() {
        host.renderSongs(host.libraryListController.filter(host.libraryState.tracks));
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/SongsMenuRenderer.kt
package com.dumuzeyn.mp3player

class SongsMenuRenderer(private val host: MainActivityCore) : MenuRenderer {
    override fun needsMiniSpacer(): Boolean = false

    override fun render() {
        host.songsRenderer.renderLibrary(
            host.libraryState.tracks,
            host.navigationState.search,
        )
    }
}
